llvmlite/tests/test_refprune.py

```diff
@@ -478,23 +481,46 @@ def test_fanout_raise_3(self):
         self.assertEqual(stats.fanout_raise, 2)
 
     fanout_raise_4 = r"""
-define i32 @main(i8* %ptr, i1 %cond) {
+define i32 @main(i8* %ptr, i1 %cond, i8** %excinfo) {
 bb_A:
     call void @NRT_incref(i8* %ptr)
     br i1 %cond, label %bb_B, label %bb_C
 bb_B:
-    ret i32 1, !ret_is_raise !0    ; BAD; all tails are raising without decref
+    ret i32 1    ; BAD; all tails are raising without decref
 bb_C:
-    ret i32 1, !ret_is_raise !0    ; BAD; all tails are raising without decref
+    ret i32 1    ; BAD; all tails are raising without decref
```

Comment:
The original intention of the test is to verify that no pruning is done when all branches are raising. I'd suggest adding the store to excinfo for these branches so `isRaising` will return True.